lib/puppet/indirector/rest.rb

```diff
@@ -39,66 +40,19 @@ def self.srv_service
     @srv_service || :puppet
   end
 
-  # The logic for server and port is kind of gross. In summary:
-  # IF an endpoint-specific setting is requested AND that setting has been set by the user
-  #    Use that setting.
-  #         The defaults for these settings are the "normal" server/masterport settings, so
-  #         when they are unset we instead want to "fall back" to the failover-selected
-  #         host/port pair.
-  # ELSE IF we have a failover-selected host/port
-  #    Use what the failover logic came up with
-  # ELSE IF the server_list setting is in use
-  #    Use the first entry - failover hasn't happened yet, but that
-  #    setting is still authoritative
-  # ELSE
-  #    Go for the legacy server/masterport settings, and hope for the best
+  # Select the server to use based on the settings configuration
+  # for this indirection, taking into account the HA server list.
   def self.server
     setting = server_setting()
-    if setting && setting != :server && Puppet.settings.set_by_config?(setting)
-      Puppet.settings[setting]
-    else
-      server = Puppet.lookup(:server) do
-        if primary_server = Puppet.settings[:server_list][0]
-          Puppet.debug "Dynamically-bound server lookup failed; using first entry"
-          primary_server[0]
-        else
-          setting ||= :server
-          Puppet.debug "Dynamically-bound server lookup failed, falling back to #{setting} setting"
-          Puppet.settings[setting]
-        end
-      end
-      server
-    end
+    Puppet::Util::Connection.determine_server(setting)
```

Comment:
Style suggestions: Lets maybe just put in server_setting directly here?  I think its just as clear
Puppet::Util::Connection.determine_server(server_setting)
But i'm happy either way